allennlp/training/initializers.py

```diff
@@ -0,0 +1,117 @@
+from typing import Dict, Callable
+import logging
+import re
+
+import torch.nn.init
+
+from allennlp.common.params import Params
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class InitializerApplicator:
+    """
+    Applies initializers to the parameters of a Module based on regex matches.
+    All parameters in the Module will be initialized.
+
+    """
+    def __init__(self, all_initializers: Dict[str, Callable[[torch.Tensor], None]] = None) -> None:
+        """
+        Parameters
+        ----------
+        all_initializers : Dict[str, Callable[[torch.Tensor], None]], optional (default = None)
+            A dictionary mapping parameter regexes to initializers to be applied to parameters
+            matching the regex. Additionally, it may include a "default" field containing a
+            initializer, which will be used in the case that the Applicator encounters a parameter
+            which does not match any of the regexes provided.
```

Comment:
You should note in here that you default to `torch.nn.init.normal` if there's no default supplied.  In fact, I'd just do that explicitly as its own argument, instead of hiding it in the dictionary.  It'd be more clear what's going on.